modules/sor/sorV2/lib/pools/v2/weighted/weightedPool.ts

```diff
@@ -1,35 +1,15 @@
-import { PrismaPoolWithDynamic } from '../../../../../../prisma/prisma-types';
-import { GqlPoolType } from '../../../../../../schema';
-import { Chain } from '@prisma/client';
-import { MathSol, WAD } from '../../utils/math';
 import { Address, Hex, parseEther } from 'viem';
-import { BigintIsh, SwapKind, Token, TokenAmount } from '@balancer/sdk';
-import { chainToIdMap } from '../../../../../network/network-config';
-import { TokenPairData } from '../../../../../pool/lib/pool-on-chain-tokenpair-data';
-import { BasePool } from '../basePool';
-
-export class WeightedPoolToken extends TokenAmount {
```

Comment:
Extracted WeightedPoolToken to its own file. Felt it would be better like this, because WeightedPoolV3 reuses it.